ArduCopter/mode_rtl.cpp

```diff
@@ -490,7 +490,7 @@ void ModeRTL::compute_return_target()
     float rtl_return_dist_cm = rtl_path.return_target.get_distance(rtl_path.origin_point) * 100.0f;
     // don't allow really shallow slopes
     if (g.rtl_cone_slope >= RTL_MIN_CONE_SLOPE) {
-        target_alt = MAX(curr_alt, MIN(target_alt, MAX(rtl_return_dist_cm*g.rtl_cone_slope, curr_alt+RTL_ABS_MIN_CLIMB)));
+        target_alt = MIN(target_alt, MAX(rtl_return_dist_cm * g.rtl_cone_slope, MAX(RTL_ALT_MIN, curr_alt + MAX(0, g.rtl_climb_min))));
```

Comment:
You could extract `MAX(RTL_ALT_MIN, curr_alt + MAX(0, g.rtl_climb_min))` out as a const float and then use it on 486 and  487.

```
// increase target to maximum of current altitude + climb_min and rtl altitude
const float min_rtl_alt = MAX(RTL_ALT_MIN, curr_alt + MAX(0, g.rtl_climb_min));
target_alt = MAX(target_alt, MAX(g.rtl_altitude, min_rtl_alt));
```

Then re use it here for:
```
        target_alt = MIN(target_alt, MAX(rtl_return_dist_cm * g.rtl_cone_slope, min_rtl_alt));
```

It reduces the number of `MAX` calls in one line making it a little easier to follow what is going on. (also saves come calculation steps)

I would also be tempted to break out the cone altitude calculation into another const float.

Reply:
Done